ci/plugins/cloudtest/hooks/pre-exit

```diff
@@ -9,23 +9,23 @@
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0.
 
-set -xeu
-
-echo "~~~ Cleaning up after cloudtest" >&2
-
-bin/ci-builder run stable kubectl --context kind-kind get pods -o name | grep -v -E 'kubernetes|minio|postgres|redpanda' | xargs -L 1 bin/ci-builder run stable kubectl --context kind-kind logs --prefix=true > kubectl-get-logs.log || true
-buildkite-agent artifact upload kubectl-get-logs.log
-
-bin/ci-builder run stable kubectl --context kind-kind get pods -o name | grep -v -E 'kubernetes|minio|postgres|redpanda' | xargs -L 1 bin/ci-builder run stable kubectl --context kind-kind logs --previous --prefix=true > kubectl-get-logs-previous.log || true
-buildkite-agent artifact upload kubectl-get-logs-previous.log
-
-bin/ci-builder run stable kubectl --context kind-kind get events > kubectl-get-events.log || true
-buildkite-agent artifact upload kubectl-get-events.log
-
-bin/ci-builder run stable kubectl --context kind-kind get all > kubectl-get-all.log || true
-buildkite-agent artifact upload kubectl-get-all.log
-
-bin/ci-builder run stable kubectl --context kind-kind describe all > kubectl-describe-all.log || true
-buildkite-agent artifact upload kubectl-describe-all.log
-
-bin/ci-builder run stable kubectl --context kind-kind delete all --all
+set -euo pipefail
+
+. misc/shlib/shlib.bash
+
+kubectl() {
+    bin/ci-builder run stable kubectl --context=kind-cloudtest "$@"
+}
+
+ci_unimportant_heading "cloudtest: Uploading logs..."
+for pod in $(kubectl get pods -o name | grep -v -E 'kubernetes|minio|cockroach|redpanda'); do
```

Comment:
`while read ...` unfortunately caused the loop to only run once, for the first container, causing incomplete logs to be preserved. The `for` loop seems to iterate properly and collect all the logs.

Reply:
I noticed that! Thanks for fixing! I thought shellcheck would complain about the `for` loop approach but it seems happy, so 👌🏽.